### PR TITLE
Feat(plugins): Add arista.avd.global_vars plugin

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

This PR introduces an Ansible vars plugin `arista.avd.global_vars` that intends to load a list of yaml file with less precedence than host and group vars. The use case is that there is a pattern to have one inventory and set of group/host vars per fabric with the need to have global vars used across the multiple fabric.

## Related Issue(s)

No issue open but trying to find a way

## Component(s) name

`plugins`

## How to test

```
EXAMPLES = """
# To enable the global_variable plugin in `ansible.cfg`
[defaults]
vars_plugins_enabled = arista.avd.global_vars, host_group_vars

[vars_global_vars]
# ARISTA_AVD_GLOBAL_VARS_PATHS can be used instead and will have precedence
paths = ../../relative/path/to/my/global/vars/file/or/dir, ../../relative/path/to/my/global/vars/file/2.yml
"""
```

Tested as integration test

### TODO

- [x] Documentation
- [x] Integration test

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    name: global_vars
+    version_added: "4.0.0"
+    short_description: Variable plugins to allow loading global_vars with less precedence than group_vars or host_vars
+    requirements:
+        - Should run at the 'inventory' stage (default) before all other
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
        - Should run at the 'inventory' stage (default) before all other variable plugins
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    description:
+        - Loads variables from a variable file present in ansible.cfg and inject them in the 'all' group
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
        - Loads variables from the variable files listed in ansible.cfg or environment variable.
        - Assign the loaded variables to the 'all' inventory group.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        required: true
+        type: list
+        elements: string
+        ini:
+          - key: paths
+            section: vars_global_vars
+        env:
+          - name: ARISTA_AVD_GLOBAL_VARS_PATHS
+        description:
+          - List of relative paths, relative  to the inventory file.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
          - List of relative paths, relative to the inventory file.
```

File: ansible_collections/arista/avd/plugins/vars/global_vars.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        default: inventory
+        choices: ["inventory"]
+        description:
+          - The stage during which executing the plugin. It could be 'inventory' or 'task'
+          - Given the expected usage of this plugin at the beginning of the run. It is hardcoded to 'inventory'
+      _valid_extensions:
+        default: [".yml", ".yaml", ".json"]
+        description:
+          - "Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these."
+          - 'This affects vars_files, include_vars, inventory and vars plugins among others.'
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/vars/global_vars.py]:
```suggestion
          - Check all of these extensions when looking for 'variable' files which should be YAML or JSON or vaulted versions of these.
          - This affects vars_files, include_vars, inventory and vars plugins among others.
```

File: ansible_collections/arista/avd/plugins/README.md [hunk 5]
```diff
@@ -377,6 +377,67 @@ switch:
                         platform_settings | selectattr("platforms", "arista.avd.contains", "default") | first) }}
 ```
 
+## Vars Plugins
+
+### arista.avd.global_vars
+
+Loads variables from the variable file present in ansible.cfg or environment variable. Assign the loaded variables to the 'all' inventory group. Files are restricted by extension to one of .yaml, .json, .yml or no extension. Hidden (starting with '.') and backup (ending with '~') files and directories are ignored. Only applies to inventory sources that are
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/README.md]:
```suggestion
Loads variables from variable files specified in ansible.cfg or in the environment variable. Assign the loaded variables to the 'all' inventory group. Files are restricted by extension to one of .yaml, .json, .yml or no extension. Hidden files (starting with '.') and backup files (ending with '~') and directories are ignored. Only applies to inventory sources that are
```

File: ansible_collections/arista/avd/plugins/README.md [hunk 6]
```diff
@@ -377,6 +377,67 @@ switch:
...
+existing paths.
+
+Should run at the 'inventory' stage (default) before all other variable plugins inject the variables before any group and host vars.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/README.md]:
```suggestion
The `arista.avd.global_vars` inventory plugin should run at the `inventory` stage (default) before all other variable plugins, to inject the variables before any group and host vars.
```

File: ansible_collections/arista/avd/plugins/README.md [hunk 7]
```diff
@@ -377,6 +377,67 @@ switch:
...
+
+**parameters:**
+
+```yaml
+- paths:
+        List of relative paths, relative to the inventory file.
+        If path is a directory, all the valid files inside are loaded in alphabetical order.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/plugins/README.md:403]: Above you say directories are ignored.

Comment (gmuloc) [reply ansible_collections/arista/avd/plugins/README.md:403]: haha good catch - there was a FIXME in my code ;) now it works!

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM, tested with customer inventory and with platform-specific variables still inherited with include_vars.

**Pull request merged**